lib/model.js

```diff
@@ -333,6 +333,13 @@ ModelBaseClass.prototype._initProperties = function(data, options) {
     if (propVal === undefined && persistUndefinedAsNull) {
       self.__data[p] = propVal = null;
     }
+    // coerce array values of primitives from string
+    if (Array.isArray(type) && type[0].name) {
+      const actualType = type[0];
+      if (BASE_TYPES[actualType.name] && Array.isArray(self.__data[p])) {
+        self.__data[p].forEach((ele, idx) => self.__data[p][idx] = new actualType(ele));
```

Comment:
`ele` sounds weird to me, can we use `elem` instead? Or even `it` (standing for _array item_).

Also I think this can be more elegantly expressed using `map` function?

```ts
self._data[p] = self._data[p].map(item => new actualType(item));
```

Reply:
> I think this can be more elegantly expressed using `map` function?
>
> ```ts
> self._data[p] = self._data[p].map(item => new actualType(item));
> ```

☝️